kie-internal/src/main/java/org/kie/internal/pmml/PMMLImplementationsUtil.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.internal.pmml;
+
+import org.kie.api.pmml.PMMLConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.kie.api.pmml.PMMLConstants.KIE_PMML_IMPLEMENTATION;
+import static org.kie.api.pmml.PMMLConstants.LEGACY;
+import static org.kie.api.pmml.PMMLConstants.NEW;
+
+/**
+ * Class used to provide utility methods to manage implementation to be invoked
+ * at runtime
+ */
+public class PMMLImplementationsUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PMMLImplementationsUtil.class);
+    private static final String LEGACY_IMPL = "org.kie.pmml.assembler.PMMLAssemblerService";
+    private static final String TRUSTY_IMPL = "org.kie.pmml.evaluator.assembler.service.PMMLAssemblerService";
+
+    /**
+     * @param classLoader
+     * @return the <b>PMML</b> implementation to enable
+     */
+    public static PMMLConstants toEnable(final ClassLoader classLoader) {
+        boolean isLegacyPresent = isImplementationPresent(LEGACY_IMPL, classLoader);
+        boolean isTrustyPresent = isImplementationPresent(TRUSTY_IMPL, classLoader);
+        if (!isLegacyPresent && !isTrustyPresent) {
+            throw new IllegalStateException("Could not find PMML implementation");
+        }
+        PMMLConstants pmmlConstants = PMMLConstants.byName(System.getProperty(KIE_PMML_IMPLEMENTATION.getName(), LEGACY.getName()));
+        switch (pmmlConstants) {
+            case LEGACY:
+                if (isLegacyPresent) {
+                    return LEGACY;
+                } else {
+                    return NEW;
```

Comment:
I'm not sure about this logic. It may happen that user specify KIE_PMML_IMPLEMENTATION, but he will get different implementation because of missing classes. I think it would be better to fail fast or at least log some warning. @gitgabrio @danielezonca  What do you think? In my opinion there are now three possible implementations and it may be some source of mistakes and confusion. Therefore I would suggest to at least every time log which implementation is used.

Reply:
@jiripetrlik 
I think to understand what you mean. Our idea about the KIE_PMML_IMPLEMENTATION flag was to use it as "last resort" only when BOTH implementations are in the classpath, that should be a transitory (I hope really transitory) situation.
As soon as possible we may remove the "legacy" implementation, and this flag with it.
I agree, anyway, to always log the implementation used in this method. @danielezonca wdyt ?

Reply:
Yes better to log the result.
Please make property win because user cannot (should not) rebuild/change BC so it will be impossible for them to revert to `kie-pmml`

Reply:
@danielezonca 
What if user insert a given KIE_PMML_IMPLEMENTATION but put the other one (or none) in the classpath ? Throw Exception ? (This is the opposite policy we agreed upon when inserted the KIE_PMML_IMPLEMENTATION flag)

Reply:
@jiripetrlik @danielezonca 
Done